fabric-data-attachment-api-v1/src/main/java/net/fabricmc/fabric/mixin/attachment/BlockEntityMixin.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.mixin.attachment;
+
+import org.spongepowered.asm.mixin.Dynamic;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Unique;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+import net.minecraft.block.entity.BlockEntity;
+import net.minecraft.nbt.NbtCompound;
+
+import net.fabricmc.fabric.api.attachment.v1.AttachmentContainer;
+import net.fabricmc.fabric.api.attachment.v1.AttachmentTarget;
+import net.fabricmc.fabric.impl.attachment.AttachmentContainerImpl;
+
+@Mixin(BlockEntity.class)
+public class BlockEntityMixin implements AttachmentTarget {
+	@Unique
+	private AttachmentContainer fabric_dataAttachments = null;
+
+	@Override
+	public AttachmentContainer getAttachmentContainer() {
+		if (fabric_dataAttachments == null) {
+			fabric_dataAttachments = AttachmentContainer.create();
+		}
+
+		return fabric_dataAttachments;
+	}
+
+	@Dynamic
+	@Inject(
+			at = @At(value = "INVOKE", target = "net/minecraft/block/entity/BlockEntity.readNbt(Lnet/minecraft/nbt/NbtCompound;)V"),
+			method = "method_17897" // lambda body in BlockEntity#createFromNbt
+	)
+	public void injectReadNbt(NbtCompound nbt, String id, BlockEntity blockEntity, CallbackInfoReturnable<BlockEntity> cir) {
```

Comment:
This fails the build.

Reply:
Yeah I haven't actually tried running this since I'm still at the prototyping stage... Ignore this for now 😅